lib/ruby_indexer/lib/ruby_indexer/visitor.rb

```diff
@@ -124,12 +132,19 @@ def visit_call_node(node)
     def visit_def_node(node)
       method_name = node.name.to_s
       comments = collect_comments(node)
-      case node.receiver
+      entry_class = case node.receiver
       when nil
-        @index << Entry::InstanceMethod.new(method_name, @file_path, node.location, comments, node.parameters)
+        if @singleton_class_node # i.e. `class << self`
+          Entry::SingletonMethod
```

Comment:
This is not necessarily true, and depends on the `class << xxx` call:
```ruby
obj = Object.new
obj.foo # NoMethodErro
class << obj
  def foo = 42
end
obj.foo # => 42
```

In this case, the method is added to the singleton class of **an instance**, so it ends up being an instance method, not a singleton class method.